pkg/secretkeys/secretkeys.go

```diff
@@ -0,0 +1,106 @@
+package secretkeys
```

Comment:
I'll come back to this file after I've looked at the wider logic.

One note, probably need some tweaks in this file (and elsewhere in the changes) to match to code format in the rest of KD. Particularly function signatures; if there's at least one function argument then they are of the form:
```
func myFuncName(
    arg1,
    arg2,
) returnType {
```
and all function signatures (whether zero or more args) have a blank line between the returnType and the start of the function body.

Reply:
Thanks for pointing it out, I didn't notice it at all.